core/xml.js

```diff
@@ -349,6 +349,14 @@ Blockly.Xml.domToWorkspace = function(xml, workspace) {
       }
     } else if (name == 'shadow') {
       goog.asserts.fail('Shadow block cannot be a top-level block.');
+    }  else if (name == 'variables') {
+      if (i == 1) {
```

Comment:
Question: should this be `i == 0` if you want `variables` to be the first child node?